### PR TITLE
Switch to GitHub actions

### PR DESCRIPTION
**Pull request by omus**
CI currently isn't running for this package. Replaces https://github.com/JuliaCollections/IterTools.jl/pull/82

### CODE REVIEW

**Comment by omus** [comment]
cc: @iamed2

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        version:
+          - "0.7"  # Earliest supported version
+          - "1.0"  # LTS
+          - "1"    # Latest release
+          - nightly
```
Comment (omus) [inline .github/workflows/CI.yml:16]: Probably not going to work. Starting with this as it was in the Travis CI config

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          julia --project=docs -e '
+            using Pkg
+            Pkg.develop(PackageSpec(path=pwd()))
+            Pkg.instantiate()'
+      - run: |
+          julia --project=docs -e '
+            using Documenter: DocMeta, doctest
+            using IterTools
+            DocMeta.setdocmeta!(IterTools, :DocTestSetup, :(using IterTools); recursive=true)
+            doctest(IterTools)'
```
Comment (omus) [inline .github/workflows/CI.yml:62]: May require a newer version of Documenter

Comment (owiecc) [reply .github/workflows/CI.yml:62]: I did some experiments on my [fork](https://github.com/owiecc/IterTools.jl/tree/CI-bump-documenter). CI failures for the docs seem to be related to the `make.jl` config. Specifically bumping Documenter.jl to 0.27 and removing some arguments helped. All the CI jobs are passing on my fork but I am not sure if I configured the Documenter correctly.

**Comment by owiecc** [comment]
I merged #82 #83 and #84 to enable CI and added some small fixes there.

**Pull request closed without merging**